src/github.com/getlantern/enproxy/conn_intf.go

```
@@ -201,18 +213,52 @@ func (c *Conn) Write(b []byte) (n int, err error) {
 
 // Read() implements the function from net.Conn
 func (c *Conn) Read(b []byte) (n int, err error) {
+	err = c.getAsyncErr()
+	if err != nil {
+		return
```

We don't want to return the error here?